js/codemeta_generation.js

```diff
@@ -312,21 +312,72 @@ function importShortOrg(fieldName, doc) {
     }
 }
 
+function importReview(doc) {
+    if (doc !== undefined) {
+        directReviewCodemetaFields.forEach(item => {
+            setIfDefined('#' + item, doc[item]);
+        });
+    }
+}
+
+function authorsEqual(author1, author2) {
+    // TODO should test more properties for equality?
+    return author1.givenName === author2.givenName
+        && author1.familyName === author2.familyName
+        && author1.email === author2.email;
+}
+
+function getSingleAuthorsFromRoles(docs) {
+    return docs.filter(doc => getDocumentType(doc) === "Role")
+        .map(doc => doc["schema:author"])
+        .reduce((authorSet, currentAuthor) => {
+            const foundAuthor = authorSet.find(author => authorsEqual(author, currentAuthor));
+            if (!foundAuthor) {
+                return authorSet.concat([currentAuthor]);
+            } else {
+                return authorSet;
+            }
+        }, []);
+}
+
+function importRoles(personPrefix, roles) {
+    roles.forEach(role => {
+        const roleId = addRole(`${personPrefix}`);
+        directRoleCodemetaFields.forEach(item => {
+            setIfDefined(`#${personPrefix}_${item}_${roleId}`, role[item]);
+        });
+    });
+}
+
 function importPersons(prefix, legend, docs) {
     if (docs === undefined) {
         return;
     }
 
-    docs.forEach(function (doc, index) {
+    const authors = docs.filter(doc => getDocumentType(doc) === "Person");
+    const authorsFromRoles = getSingleAuthorsFromRoles(docs);
+    const allAuthorDocs = authors.concat(authorsFromRoles)
+        .reduce((authors, currentAuthor) => {
+            if (!authors.find(author => authorsEqual(author, currentAuthor))) {
+                authors.push(currentAuthor);
+            }
+            return authors;
+        }, []);
```

Comment:
I wanted to point out that it may be a performance issue because it runs in quadratic time (for every author, look through the list of author). However it's fine with 1k authors, and my browser couldn't import 10k authors anyway, so I guess it's not actually a problem. (Plus, even with 100 authors it's already pretty hard to navigate the UI)

Reply:
Noted!